pkg/migration/generate.go

```diff
@@ -1152,6 +1158,17 @@ func initRuntimeOptions(cluster *corev1.StorageCluster) {
 	}
 }
 
+func getStorageClusterNameFromClusterID(clusterID string) string {
+	// Must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character
+	// If the cluster ID is a valid k8s object name, just set it as the storage cluster name,
+	// otherwise format it to be k8s compatible
+	validNameRegex := regexp.MustCompile(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`)
+	if validNameRegex.MatchString(clusterID) {
+		return clusterID
+	}
+	return pxutil.GetK8sCompatibleObjectName(clusterID)
```

Comment:
Just for migration, should we leave the hyphens? Otherwise the stc name could become unreadable.
Most people might have underscores in their names as separators. So we can replace the `_` with `-` first and then remove the rest of the unknown characters, lower case it, etc.
For instance, `System_Test_Cluster` can become `system-test-cluster` instead of `systemtestcluster`. 
Wdyt?

Reply:
I thought about it and did this originally, but I found how we process the configmap name and thought it's better to be consistent and share the same util function. removing all the special chars is more safe, and don't need to worry about other corner cases like tailing `-` or other invalid inputs. but if we don't see those corner cases I think it's good to just replace invalid chars with `-`

Reply:
The configmap case is actually different. We do it that way because portworx creates the configmaps that way. Operator just does the work of deleting them on uninstall. So we had to generate the name accordingly.

If it is not very complicated, can we replace the `_` with `-`? Asking because it would be the most common scenario.
```
name := strings.Trim(strings.ReplaceAll(clusterID, "_", "-"), "-")
// after this do what you are doing already except that just keep the hyphen in the string/regex
```